packages/nimble-components/src/select/index.ts

```diff
@@ -228,6 +229,13 @@ export class Select
         }
     }
 
+    public override disconnectedCallback(): void {
```

Comment:
Don't you need to start observing again in the connected callback? If you disconnect and then reconnect without `selectedIndexChanged` being called, then the selected option wouldn't be observed anymore, right?

Reply:
Yep, added a test for this as well.